### PR TITLE
Don't rename recovery translog in gateway

### PR DESCRIPTION
**Pull request by s1monw**
we used to reuse files / write them twice in translog. Today we don't
do this anymore so the rename logic is unneeded.

### CODE REVIEW

**Comment by s1monw** [comment]
@kimchy can you take a look

**Comment by kimchy** [comment]
LGTM, great that we don't need this logic anymore

**Comment by s1monw** [comment]
FYI - 1.x still needs this logic!

File: src/main/java/org/elasticsearch/index/gateway/IndexShardGateway.java [hunk 1]
```diff
@@ -184,36 +184,20 @@ public void recover(boolean indexShouldExists, RecoveryState recoveryState) thro
...
-                            }
-                        }
-                    } else {
-                        logger.trace("Translog file NOT found in {} - continue", translogLocation);
-                    }
-                } else {
-                    recoveringTranslogFile = tmpRecoveringFile;
-                    break;
+                // we have to support .recovering since it's a leftover from previous version but might still be on the filesystem
+                // we used to rename the foo into foo.recovering since foo was reused / overwritten but we fixed that in 1.5
```
Comment (bleskes) [inline src/main/java/org/elasticsearch/index/gateway/IndexShardGateway.java]: double checking before I change it - the comment "overwritten but we fixed that in 1.5" is wrong right? it was fixed in 2.0 as far as I can tell.

**Pull request merged**